processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/extractor/VulnerabilityNotificationExtractor.java

```diff
@@ -49,14 +48,27 @@ public VulnerabilityNotificationExtractor(Gson gson) {
     }
 
     @Override
-    public FilterableNotificationWrapper convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel, VulnerabilityNotificationContent vulnerabilityNotificationContent) {
-        List<String> affectedProjectNames = vulnerabilityNotificationContent.getAffectedProjectVersions()
-                                                .stream()
-                                                .map(AffectedProjectVersion::getProjectName)
-                                                .collect(Collectors.toList());
+    public List<FilterableNotificationWrapper> convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel, VulnerabilityNotificationContent vulnerabilityNotificationContent) {
         List<String> applicableSeverityTypes = new LinkedList<>(getApplicableSeverityTypes(vulnerabilityNotificationContent));
 
-        return FilterableNotificationWrapper.vulnerability(alertNotificationModel, vulnerabilityNotificationContent, affectedProjectNames, applicableSeverityTypes);
+        return breakDownIntoSeparateNotifications(vulnerabilityNotificationContent).stream()
+                   .map(vulnerabilitySingleProjectNotificationContent -> FilterableNotificationWrapper.vulnerability(
```

Comment:
This seems like it should be done in the original loop.

Reply:
Wow don't know how I missed that, good catch